### ISSUE TITLE
QuickGELU - not SOTA ?

### ISSUE THREAD

**Issue by Green-Sky** [monatis/clip.cpp#10]
I just read through the [open_clip readme](https://github.com/mlfoundations/open_clip#pretrained-model-interface) and found this section:
> NOTE: Many existing checkpoints use the QuickGELU activation from the original OpenAI models. This activation is actually less efficient than native torch.nn.GELU in recent versions of PyTorch. The model defaults are now nn.GELU, so one should use model definitions with -quickgelu postfix for the OpenCLIP pretrained weights. All OpenAI pretrained weights will always default to QuickGELU. One can also use the non -quickgelu model definitions with pretrained weights using QuickGELU but there will be an accuracy drop, for fine-tune that will likely vanish for longer runs.

does that mean, that we should not use QuickGELU/insert it as a hyperparameter ?

**Comment by monatis** [comment]
> does that mean, that we should not use QuickGELU

I don't think so, It was not possible to match Transformers's vectors without it with OpenAI's CLIP.

> insert it as a hyperparameter ?

Probably yes, it's defined in config.json and we can get it from there as a hyperparameter.

**Comment by monatis** [comment]
I'm taking this now
